[vcpkg baseline][treehopper] Only build dynamic libraries

FrankXie05 opened the pull request:
<!-- If your PR fixes issues, please note that here by adding "Fixes #NNNNNN." for each fixed issue on separate lines. -->
Fix [CI pipeline](https://dev.azure.com/vcpkg/public/_build/results?buildId=97165&view=results) issue:
Port `treehopper` build failed on:
```
arm-neon-android
x64-android
arm64-android
```
Error:
```
- Note: treehopper only supports dynamic library linkage. Building dynamic library.
CMake Error at scripts/cmake/vcpkg_check_linkage.cmake:25 (message):
  Refusing to build unexpected dynamic library against the static CRT.

      If this is desired, please configure your triplet to directly request this configuration.
```
This port only supports `dynamic` builds, but vcpkg `android` triplets only provides static builds. Causes check linkage conflict and failure.
So added `treehopper` to `ci.baseline.txt`.
Related code:
https://github.com/microsoft/vcpkg/blob/e85cafa30def190eca0dc1413bf8c04e434af567/ports/treehopper/portfile.cmake#L1
https://github.com/microsoft/vcpkg/blob/e85cafa30def190eca0dc1413bf8c04e434af567/triplets/arm64-android.cmake#L2
<!-- If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/ -->

<!-- If this PR updates an existing port, please uncomment and fill out this checklist:

- [ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is added to each modified port's versions file.

END OF PORT UPDATE CHECKLIST (delete this line) -->

<!-- If this PR adds a new port, please uncomment and fill out this checklist:

- [ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [ ] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [ ] The versioning scheme in `vcpkg.json` matches what upstream says.
- [ ] The license declaration in `vcpkg.json` matches what upstream says.
- [ ] The installed as the "copyright" file matches what upstream says.
- [ ] The source code of the component installed comes from an authoritative source.
- [ ] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is in the new port's versions file.
- [ ] Only one version is added to each modified port's versions file.

END OF NEW PORT CHECKLIST (delete this line) -->

FrankXie05 committed:
[vcpkg baseline][deogon/libusb]passing remove from fail

FrankXie05 committed:
[vcpkg baseline][treehopper] Only build dynamic libraries

dg0yt commented:
Come on. You document that this is not a CI problem, so don't change ci.baseline.txt. Fix the manifest instead.

FrankXie05 committed:
add libdc to ci.baseline.txt

FrankXie05 committed:
add supports

FrankXie05 committed:
version

FrankXie05 committed:
add disable static CRT

FrankXie05 committed:
version

FrankXie05 committed:
remove ci.baseline.txt

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

BillyONeal committed:
Also add treehopper

```console
PS D:\vcpkg> .\vcpkg.exe depend-info treehopper:x64-android
libusb:
vcpkg-cmake:
vcpkg-cmake-config:
treehopper: libusb, vcpkg-cmake, vcpkg-cmake-config
```

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

BillyONeal commented:
Depends on: https://github.com/microsoft/vcpkg/pull/35434

BillyONeal committed:
Merge commit '5557a33a6c4a14631b1bd65d1060a55736b1ce22' into HEAD

# Conflicts:
#	scripts/ci.baseline.txt

BillyONeal committed:
Fix android typo.

BillyONeal committed:
Merge remote-tracking branch 'FrankXie05/dev/Frank/ci_baseline' into HEAD

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

# Conflicts:
#	scripts/ci.baseline.txt

BillyONeal commented:
I am merging around build not done yet because https://dev.azure.com/vcpkg/public/_build/results?buildId=97350&view=results is building the same bits
